sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -110,18 +110,23 @@ public synchronized void setMetrics(
       return;
     }
 
-    try {
-      // NOTE: removing an activity does not reset the frame counts, only reset() does
-      frameMetricsAggregator.remove(activity);
-    } catch (Throwable ignored) {
-      // throws IllegalArgumentException when attempting to remove OnFrameMetricsAvailableListener
-      // that was never added.
-      // there's no contains method.
-      // throws NullPointerException when attempting to remove OnFrameMetricsAvailableListener and
-      // there was no
-      // Observers, See
-      // https://android.googlesource.com/platform/frameworks/base/+/140ff5ea8e2d99edc3fbe63a43239e459334c76b
-    }
+    activity.runOnUiThread(
+        () -> {
+          try {
+            // NOTE: removing an activity does not reset the frame counts, only reset() does
+            frameMetricsAggregator.remove(activity);
```

Comment:
`remove` already has a try/catch, so the problem isn't here.

Reply:
I'd bet on `stop` or `add`, both interacts with the `OnFrameMetricsAvailableListener` callback.

Reply:
It probably still is, and that's the pain of `ConcurrentModificationException`: Calling `.remove()` might works just fine on the background thread, but may crash the main thread while it's iterating the list.
But I agree, it might be worth checking other use cases as well 👍

Reply:
Yep, just added a try/catch on all operations worked on RN because we don't have control over which thread the native channels are called, here we do.
Consider using the `MainLooperHandler` and `MainThreadChecker.isMainThread()`.
See example:
https://github.com/getsentry/sentry-java/blob/4dd88fea6a1c87d5e7ad258ab82981b71328d6a5/sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java#L118-L124

Reply:
Yeah looks to me like we need a fix here that prevents a crash somewhere else. Couldn't find any mention that `FrameMetricsAggregator` must be manipulated on the UI thread only in the docs.

Calling `add` should be fine here as it's already on the UI thread because the call is coming from `ActivityLifecycleIntegration` (which implements `ActivityLifecycleCallbacks`).

For `stop` and `reset` we could also push them to the UI thread but they are much less likely to cause problems because they should only be run when the hub is closed which on Android I presume only happens if the dev manually calls it (@marandaneto please correct me here).

@markushi can you please also compare your fix with https://github.com/getsentry/sentry-java/blob/6.4.2/sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java as this is what was running when the crashes happened. Looks fine to me but would still prefer someone else to take a look.

Reply:
The other option is to make the calls to `frameMetricsAggregator` atomic, by using a `synchronized` operation, but the option above seems better to me.

Reply:
@marandaneto that wouldn't help because we don't have access to all the places where the problematic collection is being iterated.